src/react/databrowser/buckets/details/Overview.tsx

```diff
@@ -97,14 +96,43 @@ function Overview({ bucket, ingestionStates }: Props) {
   const workflowsQuery = useWorkflows();
   const features = useSelector((state: AppState) => state.auth.config.features);
   const { account } = useCurrentAccount();
+  const [isErrorModalOpen, setIsErrorModalOpen] = useState(false);
+  const [attachedWorkflowsCount, setAttachedWorkflowsCount] = useState(0);
   useEffect(() => {
     dispatch(getBucketInfo(bucket.Name));
   }, [dispatch, bucket.Name]);
 
-  const handleDeleteClick = () => {
+  const workflowAttachedError = (count: number) => (
+    <div>
+      The bucket you tried to delete has{' '}
+      {maybePluralize(count, 'workflow', 's', true)} attached to it, you should{' '}
+      <StyledLink
+        to={{
+          pathname: '/workflows',
+        }}
+      >
+        delete
+      </StyledLink>{' '}
+      {count > 1 ? 'them' : 'it'} first.
+    </div>
+  );
+
+  const handleDeleteClick = (
+    bucketName: string,
+    replicationStreams: Array<Replication>,
```

Comment:
actually we have three types of workflows: replication, expiration and transition. so we should check all of them.

Reply:
@Cuervino The pre-existing logic for preventing deletion of a bucket was specifically (and only) for attached replication workflows. Do you want me to keep it as it is, or do you want to check for expiration and transition workflows as well?

Reply:
I think we should prevent the deletion of buckets if workflows are attached for them as well, this is something we missed before :/